internal/provider/resource_network_pool.go

```diff
@@ -97,92 +146,103 @@ func ResourceNetworkPool() *schema.Resource {
 	}
 }
 
-func resourceNetworkPoolCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	apiClient := meta.(*api_client.SddcManagerClient).ApiClient
+func (r *ResourceNetworkPool) Create(ctx context.Context, req resource.CreateRequest, res *resource.CreateResponse) {
+	var data ResourceNetworkPoolModel
+
+	res.Diagnostics.Append(req.Config.Get(ctx, &data)...)
 
 	createParams := network_pools.NewCreateNetworkPoolParamsWithContext(ctx).
 		WithTimeout(constants.DefaultVcfApiCallTimeout)
-	networkPool := models.NetworkPool{}
-
-	if name, ok := d.GetOk("name"); ok {
-		networkPool.Name = name.(string)
+	networkPool := models.NetworkPool{
+		Name: data.Name.ValueString(),
 	}
 
-	if len(d.Get("network").([]interface{})) > 0 {
-		networks := d.Get("network").([]interface{})
+	var networks []NetworkModel
+	res.Diagnostics.Append(types.List.ElementsAs(data.Networks, ctx, &networks, false)...)
+
+	if len(networks) > 0 {
 		networkPool.Networks = make([]*models.Network, len(networks))
 
 		for i, network := range networks {
-			networkMap := network.(map[string]interface{})
 			networkPool.Networks[i] = &models.Network{
-				Gateway: networkMap["gateway"].(string),
-				Mask:    networkMap["mask"].(string),
-				Mtu:     int32(networkMap["mtu"].(int)),
-				Subnet:  networkMap["subnet"].(string),
-				Type:    networkMap["type"].(string),
-				VlanID:  int32(networkMap["vlan_id"].(int)),
+				VlanID:  int32(network.VlanId.ValueInt64()),
+				Gateway: network.Gateway.ValueString(),
+				Mask:    network.Mask.ValueString(),
+				Subnet:  network.Subnet.ValueString(),
+				Mtu:     int32(network.Mtu.ValueInt64()),
+				Type:    network.Type.ValueString(),
 			}
 
-			ipPools := networkMap["ip_pools"].([]interface{})
+			var ipPools []IpPoolModel
+			res.Diagnostics.Append(types.List.ElementsAs(network.IpPools, ctx, &ipPools, false)...)
 			networkPool.Networks[i].IPPools = make([]*models.IPPool, len(ipPools))
 			for j, ipPool := range ipPools {
-				ipPoolMap := ipPool.(map[string]interface{})
-
 				networkPool.Networks[i].IPPools[j] = &models.IPPool{
-					Start: ipPoolMap["start"].(string),
-					End:   ipPoolMap["end"].(string),
+					Start: ipPool.Start.ValueString(),
+					End:   ipPool.End.ValueString(),
 				}
 			}
 		}
 	}
 
 	createParams.NetworkPool = &networkPool
 
-	_, created, err := apiClient.NetworkPools.CreateNetworkPool(createParams)
+	timeout, diags := data.Timeouts.Create(ctx, 30*time.Minute)
+	res.Diagnostics.Append(diags...)
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
```

Comment:
Where does this function come from? It cancels the above context?

Reply:
Thanks for mentioning this, now I noticed that this timeout doesn't actually work.

But first to answer your question - https://developer.hashicorp.com/terraform/plugin/framework/resources/timeouts
This is how you declare a default timeout in the plugin framework. The purpose of this block is to mimic the original implementation where there's a default 30 min timeout on creating network pools.
The cancel function terminates the execution with the same error that we get today - context deadline exceeded.

This code isn't where it should be though, I'm going to move it up to the place where I create the request parameters. This timed context should be used so that the actual request has a time limit.